### PR TITLE
Realign switch styles

### PR DESCRIPTION
**Pull request by teenwolfblitzer**
Focus and hover states for the switch element vary slightly from the Figma spec as seen in the attached screen.
- Focus state outline color is sage/grey instead of sage/primary
- Hover state increases intensity of the drop shadow on the toggle control and switch body
- Add error state styling

Expected:
switch

Actual: (focus)
switch-focus-current

### CODE REVIEW

**Review by cmcfadzean — APPROVED** [review]
Like the class adds

File: app/assets/stylesheets/sage/patterns/elements/_switch.scss [hunk 1]
```diff
@@ -7,101 +7,111 @@
...
+  vertical-align: top;
+  appearance: none;
+  color: $sage-switch-color-default;
+  background: $sage-switch-color-default;
+  border-radius: $sage-switch-border-radius;
+  outline: none;
+  transition: background 0.3s ease-out;
+  cursor: pointer;
+
+  + label {
```
Comment (AndrwM) [inline app/assets/stylesheets/sage/patterns/elements/_switch.scss]: Marking this to ensure we're communicating in the docs that `<label>`s must come after `<input.sage-switch__input>` in the HTML.

Comment (philschanely) [inline app/assets/stylesheets/sage/patterns/elements/_switch.scss]: Could we implement a BEM setup on this as well?

**Review by AndrwM — APPROVED** [review]
Nice! +1

File: app/views/sage/examples/elements/switch/_preview.html.erb [hunk 2]
```diff
@@ -1,15 +1,20 @@
 <!-- Default -->
 <div class="sage-switch">
-  <input id="s1" type="checkbox">
-  <label for="s1">Switch</label>
+  <input type="checkbox" class="sage-switch__input" id="sage-switch-1" name="sage-switch-1">
+  <label for="sage-switch-1">Switch</label>
```
Comment (philschanely) [inline app/views/sage/examples/elements/switch/_preview.html.erb]: See BEM request above.

**Review by philschanely — COMMENTED** [review]
Just requesting BEM be used on label. Great otherwise!

**Review by pixelflips — APPROVED** [review]
👍 Nice work. Just a thought and by no means a blocker at all, but I noticed in the Figma file the label is also red for the error switch. I'm not even sure it even applies since switches may not have a "label" in many cases.

**Comment by teenwolfblitzer** [comment]
> 👍 Nice work. Just a thought and by no means a blocker at all, but I noticed in the Figma file the label is also red for the error switch. I'm not even sure it even applies since switches may not have a "label" in many cases.

Great callout, I've added the ability for the label to change to red as well. That being said, we'll need to do another pass for more useful form validation on the client side, since unchecked and `required` inputs here will default to appearing invalid on page load.

**Pull request merged**